roles/tasksite/tasks/main.yml

```diff
@@ -0,0 +1,203 @@
+---
+- name: Add user
+  user: name={{ tasks_user }} home=/home/{{ tasks_user }} shell=/bin/zsh groups=webusers
+
+- name: Create directory for readonly files
+  file: path=/readonly/{{ tasks_user }} state=directory owner={{ tasks_user }} group={{ tasks_group }} mode=0750
+
+- name: Create needed private folder structure
+  file: path={{ item }} state=directory recurse=yes owner={{ tasks_user }} group={{ tasks_group }} mode=0700
+  with_items:
+  - /home/{{ tasks_user }}/.config/etc/php
+  - /home/{{ tasks_user }}/.config/etc/mysql
+  - /home/{{ tasks_user }}/.config/letsencrypt
+  - /home/{{ tasks_user }}/.local/bin
+  - /home/{{ tasks_user }}/.var/lib/mysql
+  - /home/{{ tasks_user }}/.var/log
+  - /home/{{ tasks_user }}/.var/log/mysql
+  - /home/{{ tasks_user }}/.var/repo
+  - /home/{{ tasks_user }}/.var/run
+  - /home/{{ tasks_user }}/.var/storage
+  - /home/{{ tasks_user }}/.cache
+
+- name: Create needed folder structure
+  file: path={{ item }} state=directory recurse=yes owner={{ tasks_user }} group={{ tasks_group }} mode=0755
+  with_items:
+  - /var/www/{{ tasks_user }}/domains
+  - /var/www/{{ tasks_user }}/cgi-bin
+  - /var/www/{{ tasks_user }}/fcgi-bin
+  - /var/www/{{ tasks_user }}/letsencrypt
+  - /var/www/{{ tasks_user }}/opt/phabricator
+
+- name: Create directories for Apache log files and SSL certificates
+  file: path=/readonly/{{ tasks_user }}/{{ item }} state=directory owner=root group=root mode=0755
+  with_items:
+    - log
+    - ssl
+
+- name: Install CGI wrapper
+  template: src=php-fcgi-starter dest=/var/www/{{ tasks_user }}/fcgi-bin/ mode=700 owner={{ tasks_user }} group={{ tasks_group }}
+
+- name: Install necessary packages
+  apt: name={{ item }}
+  with_items:
+    - python-mysqldb
+    - python-pygments
+
+
+# MySQL
+- name: Generate MySQL password
+  shell: head /dev/urandom | tr -dc A-Za-z0-9 | head -c 20 > /home/{{ tasks_user }}/.var/mysql-password
+  args:
+    creates: /home/{{ tasks_user }}/.var/mysql-password
+
+- name: Restrict permissions for MySQL password file
```

Comment:
Das sollte nicht erst nachdem das Passwort rein geschrieben wurde geschehen. Dass jemand in der "kurzen Zeit" dazwischen die Datei liest ist zwar erstmal unwahrscheinlich, Ansible könnte aber z.B. wegen irgendeines Fehlers das Passwort reinschreiben, den Task trotzdem als failed ansehen und dann dieses Playbook abbrechen, wodurch die Datei bis zum korrekten Run dieses Playbooks ungeschützt wäre (aber das Passwort nicht neu erstellt würde).